src/librustc_typeck/check/mod.rs

```diff
@@ -4243,9 +4261,17 @@ fn check_expr_with_unifier<'a, 'tcx, F>(fcx: &FnCtxt<'a, 'tcx>,
                     fcx.tcx()
                        .sess
                        .span_err(path.span,
+                       // Errors may be multi-line. An aligned multi-line error will have
+                       // `expected` and `found` on consecutive lines. See issue 19464. A
+                       // newline must be prepended so `expected` is located properly.
                                  format!("structure constructor specifies a \
                                          structure of type `{}`, but this \
-                                         structure has type `{}`: {}",
+                                         structure has type `{}`:\n {}",
```

Comment:
Ah, I didn't have to give the entire error. Just a substring of the error because the regex is only looking for a match. That error is really long and I thought it might hit the `make check` parsing width limit. So I trimmed it. The full script is here at [playpen](http://is.gd/VpHO0R). The main part of the error which I pasted below:
```rust
<anon>:8:14: 8:20 error: structure constructor specifies a structure of type `Point<f32>`, but this structure has type `Point<int>`: expected f32, found int
<anon>:8     let pt = PointF {
```